util/testutil/shell.go

```diff
@@ -143,38 +181,37 @@ func (m *RemoteSnapshotMonitor) CheckAllRemoteSnapshots(t *testing.T) {
 	}
 }
 
-// FatalMonitor scans logs streamed from the specified io.Reader for fatal errors during startup
-func FatalMonitor(r shell.Reporter, stdout, stderr io.Reader) {
+// LogConfirmStartup registers a LogMonitor function to scan until startup succeeds or fails
+func LogConfirmStartup(m *LogMonitor) error {
 	errs := make(chan error, 1)
-	go scanFatalLog(io.TeeReader(stdout, r.Stdout()), errs)
-	go scanFatalLog(io.TeeReader(stderr, r.Stderr()), errs)
+	m.Add("startup", monitorStartup(errs))
```

Comment:
Why is `RemoteSnapshotMonitor` a separete struct and `monitorStartup` just a function?

Reply:
Per code review meeting, I expect to have a better answer to this question after implementing 1-2 more impending uses cases for this framework.